lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -1378,21 +1380,24 @@ selectTxHistory ti wid minWithdrawal order conditions = do
     selectLatestCheckpoint wid >>= \case
         Nothing -> pure []
         Just cp -> do
-            minWithdrawalFilter <- case minWithdrawal of
-                Nothing  -> pure []
+            let txMetaFilter = (TxMetaWalletId ==. wid):conditions
+            metas <- case minWithdrawal of
+                Nothing -> fmap entityVal <$> selectList txMetaFilter sortOpt
                 Just inf -> do
                     let coin = W.Coin $ fromIntegral $ getQuantity inf
-                    wdrls <- fmap entityVal
-                        <$> selectList [ TxWithdrawalAmount >=. coin ] []
-                    pure [ TxMetaTxId <-. (txWithdrawalTxId <$> wdrls) ]
-
-            metas <- fmap entityVal <$> selectList
-                ( mconcat
-                    [ [ TxMetaWalletId ==. wid ]
-                    , minWithdrawalFilter
-                    , conditions
-                    ]
-                ) sortOpt
+                    let sortTxId = case order of
+                            W.Ascending -> Desc TxWithdrawalTxId
+                            W.Descending -> Asc TxWithdrawalTxId
```

Comment:
I am not sure to get the rationale here, surely the extra ordering of the txid has nothing to do with the slot order? So we could pick an arbitrary direction regardless of `order`'s value, right ?